src/nvim/sign.c

```diff
@@ -1758,34 +993,26 @@ static char *get_nth_sign_group_name(int idx)
 {
   // Complete with name of sign groups already defined
   int current_idx = 0;
-  int todo = (int)sg_table.ht_used;
-  for (hashitem_T *hi = sg_table.ht_array; todo > 0; hi++) {
-    if (!HASHITEM_EMPTY(hi)) {
-      todo--;
-      if (current_idx++ == idx) {
-        signgroup_T *const group = HI2SG(hi);
-        return group->sg_name;
-      }
+  for (size_t i = 0; i < kv_size(sign_ns); i++) {
```

Comment:
this can just be a check that `idx` is in bounds `idx < kv_size(sign_ns)` and then use `kv_A(sign_ns, idx)`